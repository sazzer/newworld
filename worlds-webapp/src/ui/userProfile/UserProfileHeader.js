// @flow

import React from 'react';
import {Header, Icon} from 'semantic-ui-react'
import type {User} from "../../users/users";

/** The flow type representing the props for the User Header */
type UserProfileHeaderProps = {
    user: User
};

/**
 * The header to display for the user profile
 */
export default function UserProfileHeader(props: UserProfileHeaderProps) {
    return (
        <Header as='h2' dividing>
            <Icon name='user' />
            <Header.Content>
                { props.user.displayName }
            </Header.Content>
        </Header>
    );
}
