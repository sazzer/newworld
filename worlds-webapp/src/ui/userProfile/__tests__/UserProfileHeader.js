// @flow

import React from 'react';
import {shallow} from 'enzyme';
import UserProfileHeader  from '../UserProfileHeader';

/** Create the component to test */
function setup({user}) {
    return {
        element: shallow(<UserProfileHeader user={user} />)
    };
}

describe('UserProfileHeader', () => {
    it('Renders correctly', () => {
        const user = {
            id: 'id',
            displayName: 'Test User',
            username: 'testuser'
        };

        const { element } = setup({user});

        expect(element).toMatchSnapshot();
    });
});
