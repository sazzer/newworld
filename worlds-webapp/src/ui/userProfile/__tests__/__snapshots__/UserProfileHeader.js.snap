// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfileHeader Renders correctly 1`] = `
<Header
  as="h2"
  dividing={true}
>
  <Icon
    as="i"
    name="user"
  />
  <HeaderContent>
    Test User
  </HeaderContent>
</Header>
`;
